### PR TITLE
Add PromotionalOfferSucceededHandler with purchase data

### PR DESCRIPTION
**Pull request by vegaro**
Add new `onCustomerCenterPromotionalOfferSucceeded` modifier that provides `CustomerInfo`, `StoreTransaction`, and offer ID. Deprecate the old `onCustomerCenterPromotionalOfferSuccess` void handler. The new handler fires the deprecated publisher for backward compatibility.

---

> [!NOTE]
> **Medium Risk**
> Adds a new public callback surface and threads `CustomerInfo`/`StoreTransaction` through multiple layers (SwiftUI environment, Combine wrapper, UIKit delegate). Risk is mainly around event firing/backwards compatibility, especially when StoreKit returns a nil transaction.
> 
> **Overview**
> Adds a new Customer Center promotional offer completion callback, `onCustomerCenterPromotionalOfferSucceeded`, that provides `CustomerInfo`, `StoreTransaction`, and `offerId`, and wires it through the environment (`CustomerCenterExternalActions`), `CustomerCenterActionWrapper` publishers, SwiftUI view modifiers, and the UIKit `CustomerCenterViewController`/delegate.
> 
> Deprecates the legacy void-based `onCustomerCenterPromotionalOfferSuccess`/delegate method but keeps backward compatibility by continuing to fire the deprecated publisher on success; adds explicit handling + logging for the edge case where StoreKit returns a successful result with a nil transaction.
> 
> Updates tests and the PaywallsTester sample to cover/illustrate the new handler, including verification that the new event also triggers the deprecated callback.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 3d542905145034b459672c24e06af33cf80e33bc. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: RevenueCatUI/CustomerCenter/ViewModels/PromotionalOfferViewModel.swift [hunk 1]
```diff
@@ -77,7 +77,11 @@ final class PromotionalOfferViewModel: ObservableObject {
             } else {
                 let transactionId = result.transaction?.transactionIdentifier ?? "-"
                 Logger.debug(Strings.promo_offer_purchase_succeeded(productId, offerId, transactionId))
-                self.actionWrapper.handleAction(.promotionalOfferSuccess)
+                if let transaction = result.transaction {
+                    self.actionWrapper.handleAction(
+                        .promotionalOfferSucceeded(result.customerInfo, transaction, offerId)
+                    )
```
Comment (facumenzella) [inline RevenueCatUI/CustomerCenter/ViewModels/PromotionalOfferViewModel.swift:85]: This changes the behavior for the deprecated handler: on `main`, `.promotionalOfferSuccess` fires unconditionally on a non-cancelled purchase. Now it only fires when `transaction` is non-nil (since the backward-compat `promotionalOfferSuccess.send()` lives inside the `.promotionalOfferSucceeded` case in the action wrapper).

Apps using the deprecated `onCustomerCenterPromotionalOfferSuccess` would silently stop receiving callbacks when `transaction` is nil. Is that intentional? If so, worth calling it out in the deprecation message. If not, might want to keep firing the old handler unconditionally:

```swift
if let transaction = result.transaction {
    self.actionWrapper.handleAction(
        .promotionalOfferSucceeded(result.customerInfo, transaction, offerId)
    )
} else {
    self.actionWrapper.promotionalOfferSuccess.send(())
}
```

File: RevenueCatUI/CustomerCenter/Data/CustomerCenterActionViewModifier.swift [hunk 2]
```diff
@@ -88,6 +88,10 @@ struct CustomerCenterActionViewModifier: ViewModifier {
             .sink { actions.promotionalOfferSuccess() }
             .store(in: &cancellables)
 
+        actionWrapper.promotionalOfferSucceededPublisher
+            .sink { actions.promotionalOfferSucceeded($0.0, $0.1, $0.2) }
```
Comment (facumenzella) [inline RevenueCatUI/CustomerCenter/Data/CustomerCenterActionViewModifier.swift]: nit: the `$0.0, $0.1, $0.2` pattern is a bit fragile — any reordering of the tuple would break this silently. Worth destructuring like the delegate sink does?

```swift
.sink { customerInfo, transaction, offerId in
    actions.promotionalOfferSucceeded(customerInfo, transaction, offerId)
}
```

Same applies to the ViewController sink at line 199.

**Comment by facumenzella** [comment]
**Hidden coupling: `SubscriptionDetailViewModel` silently depends on deprecated publisher**

`SubscriptionDetailViewModel.didAppear()` subscribes only to `promotionalOfferSuccessPublisher` to drive `refreshPurchase()`. After this PR, the normal success path dispatches `.promotionalOfferSucceeded`, which happens to also fire `promotionalOfferSuccess.send(())` as a backward-compat side effect.

There's no test covering this path, and no comment documenting the dependency. If a future cleanup removes the deprecated publisher side-effect, subscription detail will silently stop refreshing after a promo-offer redemption.

Consider either:
- Subscribing `SubscriptionDetailViewModel` to `promotionalOfferSucceededPublisher` directly
- Or at minimum adding a test + comment documenting why the deprecated publisher must keep firing for internal consumers

File: RevenueCatUI/CustomerCenter/Views/UIKit Compatibility/CustomerCenterViewController.swift [hunk 3]
```diff
@@ -337,6 +347,18 @@ public class CustomerCenterViewController: UIViewController {
...
             .store(in: &cancellables)
+
+        actionWrapper.promotionalOfferSucceededPublisher
+            .sink { [weak self] customerInfo, transaction, offerId in
+                guard let self else { return }
+                self.delegate?.customerCenterViewController?(
+                    self,
+                    didSucceedWithPromotionalOffer: offerId,
+                    customerInfo: customerInfo,
+                    transaction: transaction
```
Comment (facumenzella) [inline RevenueCatUI/CustomerCenter/Views/UIKit Compatibility/CustomerCenterViewController.swift:358]: **Deprecated delegate method fires alongside new method**

In `setupDelegateBindings`, both `promotionalOfferSuccessPublisher` and `promotionalOfferSucceededPublisher` are independently subscribed to their respective delegate methods. Because `handleAction(.promotionalOfferSucceeded(...))` sends on both subjects, a delegate implementing both the deprecated and new methods will receive two callbacks for a single successful purchase.

This could cause duplicate side-effects for consumers migrating between APIs. Consider guarding the deprecated delegate call when the delegate also responds to the new selector, or documenting the dual-fire behavior explicitly in the protocol.

Comment (vegaro) [reply RevenueCatUI/CustomerCenter/Views/UIKit Compatibility/CustomerCenterViewController.swift:358]: hmm I think the alternative, checking if the delegate is implemented would be weird as well? someone might be wondering why one of the delegates is not being called
